actuator/configuration_test.go

```diff
@@ -0,0 +1,79 @@
+package actuator_test
+
+import (
+	"os"
+	"testing"
+
+	"github.com/ninech/actuator/actuator"
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/assert"
+)
+
+// LoadConfiguration
+
+func TestLoadConfiguration(t *testing.T) {
+	assert.Nil(t, actuator.LoadConfiguration())
+}
+
+// LoadConfigFile
+
+func TestLoadConfigFileNotFound(t *testing.T) {
+	config := actuator.Configuration{}
+	assert.Nil(t, config.LoadConfigFile(afero.NewMemMapFs(), &config))
+}
+
+func TestLoadConfigFileValidFile(t *testing.T) {
+	fs := afero.NewMemMapFs()
+	file, _ := fs.Create(actuator.ConfigFileName)
+	file.WriteString("---\ngithub_webhook_secret: abcd")
+
+	config := actuator.Configuration{}
+	err := config.LoadConfigFile(fs, &config)
+
+	assert.Nil(t, err)
+	assert.Equal(t, "abcd", config.GithubWebhookSecret)
+}
+
+func TestLoadConfigFileInvalidYaml(t *testing.T) {
+	fs := afero.NewMemMapFs()
+	file, _ := fs.Create(actuator.ConfigFileName)
+	file.WriteString("1234")
+
+	config := actuator.Configuration{}
+	err := config.LoadConfigFile(fs, &config)
+
+	assert.NotNil(t, err)
+}
+
+func TestLoadConfigFileRepositories(t *testing.T) {
+	fs := afero.NewMemMapFs()
+	file, _ := fs.Create(actuator.ConfigFileName)
+	file.WriteString(`---
```

Comment:
Couldn't we load a file instead of hardcoding it here?

Reply:
We could, but:
1. It would be more complicated.
2. When we add other tests which require different data in the file, we would need to have more files. Here it's very obvious what the file should contain and what's expected.